debugger/src/peripherals/svd_variables.rs

```diff
@@ -0,0 +1,347 @@
+                field_variable.memory_location = register_variable.memory_location.clone();
+                // For SVD fields, we overload the range_lower_bound and range_upper_bound as the bit range LSB and MSB.
+                field_variable.range_lower_bound = field.bit_offset() as i64;
+                field_variable.range_upper_bound = (field.bit_offset() + field.bit_width()) as i64;
+                if register_has_restricted_read {
+                    register_variable.set_value(probe_rs::debug::VariableValue::Error(
+                        "Register access doesn't allow reading, or will have side effects."
+                            .to_string(),
+                    ));
+                } else if field.read_action.is_some()
+                    || (if let Some(field_access) = field.access {
+                        field_access == Access::ReadWriteOnce || field_access == Access::WriteOnly
+                    } else {
+                        false
+                    })
+                {
+                    field_variable.set_value(probe_rs::debug::VariableValue::Error(
+                        "Field access doesn't allow reading, or will have side effects."
+                            .to_string(),
+                    ));
+                    // If we can't read any of the bits, then don't read the register either.
+                    register_variable.set_value(probe_rs::debug::VariableValue::Error(
+                        "Some fields' access doesn't allow reading, or will have side effects."
+                            .to_string(),
+                    ));
+                    register_has_restricted_read = true;
+                    register_variable = svd_cache.cache_variable(
+                        Some(peripheral_variable.variable_key),
+                        register_variable,
+                        core,
+                    )?;
+                }
+                // TODO: Extend the Variable definition, so that we can resolve the EnumeratedValues for fields.
+                svd_cache.cache_variable(
+                    Some(register_variable.variable_key),
+                    field_variable,
+                    core,
+                )?;
+            }
+        }
+    }
+
+    Ok(svd_cache)
+}
+
+/// Resolve all the peripherals through their (optional) `derived_from` peripheral.
+pub(crate) fn resolve_peripherals(
+    peripheral_device: &Device,
+) -> Result<Vec<PeripheralInfo>, DebuggerError> {
+    let mut resolved_peripherals = vec![];
+    for device_peripheral in &peripheral_device.peripherals {
+        if let Some(derived_from) = &device_peripheral.derived_from {
+            if let Some(derived_result) = peripheral_device.get_peripheral(derived_from) {
+                match &device_peripheral.derive_from(derived_result) {
+                    svd_rs::MaybeArray::Single(derived_peripheral) => {
+                        resolved_peripherals.push(derived_peripheral.clone());
```

Comment:
`&` in match and `clone` looks redundant

Reply:
Good catch. A leftover from hacking my previous code with the new derive_from() :)